### PR TITLE
Fix events being dropped when flushing if there are corrupted items in the event queue

### PR DESCRIPTION
**Pull request by zihejia**
- Fix events being dropped when flushing if there are corrupted items in the event queue (filter out the bad event rather than dropping the whole batch of queue)
- safe guard the invalid numbers from being infinite or NaN before sending to Mixpanel

### CODE REVIEW

File: MixpanelDemo/MixpanelDemo/TrackingViewController.swift [hunk 1]
```diff
@@ -45,11 +46,12 @@ class TrackingViewController: UIViewController, UITableViewDelegate, UITableView
...
         switch indexPath.item {
         case 0:
-            let ev = "Track Event!"
+            eventNo += 1
+            let ev = "Track Event! No: \(eventNo)"
             Mixpanel.mainInstance().track(event: ev)
             descStr = "Event: \"\(ev)\""
         case 1:
-            let ev = "Track Event With Properties!"
+            let ev = "Track Event With Properties! No: \(eventNo)"
```
Comment (jaredmixpanel) [inline MixpanelDemo/MixpanelDemo/TrackingViewController.swift:54]: in this example, I would make the event number a property and keep the event name consistent

Comment (zihejia) [reply MixpanelDemo/MixpanelDemo/TrackingViewController.swift:54]: oops, thx for catching this

File: Sources/MixpanelInstance.swift [hunk 2]
```diff
@@ -82,7 +82,7 @@ open class MixpanelInstance: CustomDebugStringConvertible, FlushDelegate, AEDele
     open var trackAutomaticEventsEnabled: Bool? = nil
 
     /// Flush timer's interval.
-    /// Setting a flush interval of 0 will turn off the flush timer.
+    /// Setting a flush interval of 0 will turn off the flush timer and you need call flush API manually to upload queued data to the Mixpanel server.
```
Comment (jaredmixpanel) [inline Sources/MixpanelInstance.swift:85]:
```suggestion
    /// Setting a flush interval of 0 will turn off the flush timer and you need to call the flush() API manually to upload queued data to the Mixpanel server.
```

**Review by jaredmixpanel — APPROVED** [review]
lgtm

**Pull request merged**